sycl/source/detail/platform_impl.cpp

```diff
@@ -247,47 +247,40 @@ static void filterAllowList(vector_class<RT::PiDevice> &PiDevices,
         if (!std::regex_match(PlatformName, std::regex(Desc.PlatName))) {
           PlatNameState = MatchState::NOMATCH;
           continue;
-        } else {
-          PlatNameState = MatchState::MATCH;
         }
       }
 
       if (!Desc.PlatVer.empty()) {
         if (!std::regex_match(PlatformVer, std::regex(Desc.PlatVer))) {
           PlatVerState = MatchState::NOMATCH;
           continue;
-        } else {
-          PlatVerState = MatchState::MATCH;
         }
       }
 
       if (!Desc.DevName.empty()) {
         if (!std::regex_match(DeviceName, std::regex(Desc.DevName))) {
           DevNameState = MatchState::NOMATCH;
           continue;
-        } else {
-          DevNameState = MatchState::MATCH;
         }
       }
 
       if (!Desc.DevDriverVer.empty()) {
         if (!std::regex_match(DeviceDriverVer, std::regex(Desc.DevDriverVer))) {
           DevVerState = MatchState::NOMATCH;
           continue;
-        } else {
-          DevVerState = MatchState::MATCH;
         }
       }
 
       PiDevices[InsertIDx++] = Device;
       break;
     }
   }
-  if (DevNameState == MatchState::MATCH && DevVerState == MatchState::NOMATCH) {
+  if (DevNameState == MatchState::NOMATCH &&
+      DevVerState == MatchState::NOMATCH) {
     throw sycl::runtime_error("Requested SYCL device not found",
```

Comment:
I made changes to make config/select_device.cpp LIT test pass. But I'm not sure if we need this exception at all.  @glyons-intel, could you please explain, why the exception was added to filterAllowList? Mb it should be removed?

Reply:
If the user provided the name and the driver version, then both should match.  In the code above, the name matched, but the version did not, so the device was not found.   However, at the time that I wrote this, OpenCL GPU and Level Zero GPU had different names and different versions. 

If I understand you correctly, the code matches the when it finds the first device, and then checks the second device and fails.   That is not good.  Rather than the change you are suggesting, I think it would be better to move the two state test "if" clauses inside of the loop before the device is inserted on PiDevices[].  Remove the "throw" from the two if clauses.    Only add the device to PiDevices{} if the name and version match.  If only the name is provided, just match the name.

Does this make sense?

Reply:
LGTM, thanks! I applied your suggestion and extend it to the use case where SYCL_DEVICE_ALLOWLIST has only DriverVersion or only PlatfromVersion. Does it make sense?
Now, if no devices selected based on a value from SYCL_DEVICE_ALLOWLIST, exception _No device of requested type available_ will be raised from device_selector.cpp

Reply:
yes, it makes sense.  thanks.